test/eigen.jl

```diff
@@ -75,24 +75,31 @@ using StaticArrays, Test, LinearAlgebra
         @test vals::SVector ≈ sort(m_d)
         @test eigvals(m_c) ≈ sort(m_d)
         @test eigvals(Hermitian(m_c)) ≈ sort(m_d)
+    end
 
-        # issue #523
-        for (i, j) in ((1, 2), (2, 1)), uplo in (:U, :L)
-            A = SMatrix{2,2,Float64}((i, 0, 0, j))
-            E = eigen(Symmetric(A, uplo))
-            @test eigvecs(E) * SDiagonal(eigvals(E)) * eigvecs(E)' ≈ A
-        end
-
-        m1_a = randn(2,2)
-        m1_a = m1_a*m1_a'
-        m1 = SMatrix{2,2}(m1_a)
-        m2_a = randn(2,2)
-        m2_a = m2_a*m2_a'
-        m2 = SMatrix{2,2}(m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(m1, m2)) ≈ eigvals(m1_a, m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(Symmetric(m1), Symmetric(m2))) ≈ eigvals(Symmetric(m1_a), Symmetric(m2_a))
+    # issue #523, #694
+    zero = 0.0
+    smallest_non_zero = nextfloat(zero)
+    smallest_normal = floatmin(zero)
+    largest_subnormal = prevfloat(smallest_normal)
+    epsilon = eps(1.0)
+    one_p_epsilon = 1.0 + epsilon
```

Comment:
Quibble: Better written `nextfloat(1.0)` — it's the same in this case, but if you'd written `1.0 - epsilon` that's not equal to `prevfloat(1.0)` which is a bit of a gotcha given that the boundary in floating point discretization density lies on the powers of two.

Reply:
:thumbsup: